src/appleseed/renderer/kernel/lighting/lightpathstream.cpp

```diff
@@ -267,9 +267,19 @@ void LightPathStream::create_path_from_hit_emitter(const size_t emitter_event_in
             reflector_vertex.m_radiance = current_radiance;
             m_vertices.push_back(reflector_vertex);
 
+            // If the hit is an emitter, add the radiance to current
+            if (event.m_type == EventType::HitEmitter)
```

Comment:
Wouldn't it be better to have this whole addition part after calculating the reflectance? I.e. add the emitting light last.

Reply:
Not sure what you mean here @LZaw?

Reply:
Sorry, the line of code doesn't fit the comment very well. I should have positioned it further down, my bad.
In the if statement, I marked (L271-L275) the emitter addition happens. But a bit further down (afterwards), the reflectance of the current surface is being considered (L277-L284). This means, in the current version, the emitting light is being added and then reduced by the reflection value of its own surface.
I suggest to first reduce the incoming light by the reflectance of the surface and add the emitted light then. (Simply speaking, an emitter doesn't get affected by its own reflectance value.)

Reply:
Yea myself and @LZaw were discussing this in DMs and I'm pretty sure having it at the end of the function (as it is in the commit I just pushed) is correct.